src/Symfony/Component/DependencyInjection/Container.php

```diff
@@ -195,49 +197,49 @@ public function get(string $id, int $invalidBehavior = self::EXCEPTION_ON_INVALI
     {
         return $this->services[$id]
             ?? $this->services[$id = $this->aliases[$id] ?? $id]
-            ?? ('service_container' === $id ? $this : ($this->factories[$id] ?? $this->make(...))($id, $invalidBehavior));
+            ?? ('service_container' === $id ? $this : ($this->factories[$id] ?? self::$make ??= self::make(...))($this, $id, $invalidBehavior));
     }
 
     /**
      * Creates a service.
      *
      * As a separate method to allow "get()" to use the really fast `??` operator.
      */
-    private function make(string $id, int $invalidBehavior)
+    private static function make($container, string $id, int $invalidBehavior)
```

Comment:
the type of `$container` should be defined (either though an actual type or through phpdoc at least) so that we don't loose the benefits of static analysis on that code.

Reply:
Who needs to analyze the dumped factories statically?

Reply:
it is not about the dumped factories. It is about the body of that method, where our Psalm-based CI job might detect mistakes if it knows the type of the variable.

Reply:
got it, fixed in 02613dd455